website/cue/reference/releases/0.51.0.cue

```diff
@@ -0,0 +1,478 @@
+package metadata
+
+releases: "0.51.0": {
+	date:     "2025-11-03"
+	codename: ""
+
+	whats_next: []
+
+	changelog: [
+		{
+			type: "feat"
+			description: """
+				Added `truncate` options to `file` sink to truncate output files after some time.
+				"""
+			contributors: ["esensar", "Quad9DNS"]
+		},
+		{
+			type: "feat"
+			description: """
+				Disable ANSI color for `vector test` when running non-interactively. Honor `--color {auto|always|never}` and `VECTOR_COLOR`; VRL diagnostics no longer include ANSI sequences when color is disabled.
+				"""
+			contributors: ["VanjaRo"]
+		},
+		{
+			type: "feat"
+			description: """
+				Adds proper support for compression of HEC indexer ack queries, using the sink's configured `compression` setting.
+				"""
+			contributors: ["sbalmos"]
+		},
+		{
+			type: "feat"
+			description: """
+				Added `expired` output to the memory enrichment table source, to export items as they expire in the cache.
+				"""
+			contributors: ["esensar", "Quad9DNS"]
+		},
+		{
+			type: "feat"
+			description: """
+				On receiving SIGHUP Vector now also reloads transform components with external VRL files.
+				"""
+			contributors: ["nekorro"]
+		},
+		{
+			type: "fix"
+			description: """
+				The configuration watcher now collects event paths even during the delay period. These were previously ignored and prevented components from reloading.
+				"""
+			contributors: ["nekorro"]
+		},
+		{
+			type: "fix"
+			description: """
+				Enable unused TLS settings for perform client authentication by SSL certificate in `mqtt` sink.
+				"""
+			contributors: ["ValentinChernovNTQ"]
+		},
+		{
+			type: "fix"
+			description: """
+				Memory enrichment table's outputs are now visible to the `vector tap` command.
+				"""
+			contributors: ["esensar", "Quad9DNS"]
+		},
+		{
+			type: "fix"
+			description: """
+				Fixed a panic in the `azure_blob` sink by enabling a missing required crate feature.
+				"""
+			contributors: ["thomasqueirozb"]
+		},
+		{
+			type: "fix"
+			description: """
+				Fix buffer counter underflowed, caused by the counter has not been updated(increase) timely when new event is coming.
```

Comment:
```suggestion
				Fixed an issue where the buffer counter underflowed. This was caused by the counter not being increased before a new event was observed.
```

@pront you have more context on this issue. Please fix this if I'm butchering the explanation